HashTagTokenizer.java

```diff
@@ -4,23 +4,30 @@ public class HashTagTokenizer {
 
 	public static void main(String[] args) {
 
+
 		String hashTag = args[0];
 		String []dictionary = readDictionary("dictionary.txt");
 		breakHashTag(hashTag, dictionary);
 	}
 
 	public static String[] readDictionary(String fileName) {
 		String[] dictionary = new String[3000];
-
 		In in = new In(fileName);
-
-		// Your code here
+		for (int i = 0 ; i < dictionary.length ; i++) {
+			dictionary[i] = in.readString();
+		}
 
 		return dictionary;
 	}
 
 	public static boolean existInDictionary(String word, String []dictionary) {
-		// Your code here
+		boolean exist = false;
+		for (int i = 0; i < dictionary.length; i++) {
+            if (word.equals(dictionary[i])) {
+				exist = true;
+			}
+		}
+			return exist;
```

Comment:
When the word found you for sure don't have to do any other calculation, in your code you will iterate over all dictionary even when the result already known.
```suggestion
		for (int i = 0; i < dictionary.length; i++) {
            if (word.equals(dictionary[i])) {
				return true;
			}
		}
			return false;
```